client-local/src/main/java/com/cloudera/livy/client/local/LocalClient.java

```diff
@@ -248,76 +251,60 @@ public void run() {
       // mode, the driver options need to be passed directly on the command line. Otherwise,
       // SparkSubmit will take care of that for us.
       String master = conf.get("spark.master");
+      LOG.info("Master : " + master);
+      launcher.setMaster(master);
       Preconditions.checkArgument(master != null, "spark.master is not defined.");
 
-      List<String> argv = Lists.newArrayList();
-
-      if (sparkHome != null) {
-        argv.add(new File(sparkHome, "bin/spark-submit").getAbsolutePath());
-      } else {
-        LOG.info("No spark.home provided, calling SparkSubmit directly.");
-        argv.add(new File(System.getProperty("java.home"), "bin/java").getAbsolutePath());
-
         if (master.startsWith("local") ||
               master.startsWith("mesos") ||
               master.endsWith("-client") ||
               master.startsWith("spark")) {
           String mem = conf.get("spark.driver.memory");
-          if (mem != null) {
-            argv.add("-Xms" + mem);
-            argv.add("-Xmx" + mem);
-          }
+          launcher.setConf("spark.driver.memory",mem);
```

Comment:
Also, if you notice a recent change that I made, in particular this line earlier in the file:

    File sparkConf = writeConfToFile(true);

All the spark configuration is saved to a file and should be sent to Spark by calling `SparkLauncher.setPropertiesFile`. All of this code can then go away, because the launcher code already does all this parsing and translation of options.